### PR TITLE
[ML] Clean up for Custom Applications on Compute Instances

### PR DESCRIPTION
**Pull request by nthandeMS**
# Description

- Adds ImageSettings, EndpointsSettings, VolumeSettings classes to public namespace.
  - Adds missing SetupScripts, ScriptReference classes to public namespace.
- Update Compute Instance docstring to specify Custom Applications is Experimental.
- Adds custom_applications to the constructor signature for Compute Instance.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ class ComputeInstance(Compute):
     :type enable_node_public_ip: Optional[bool], optional
     :param setup_scripts: Details of customized scripts to execute for setting up the cluster.
     :type setup_scripts: Optional[SetupScripts], optional
-    :param custom_applications: Describes available applications and their endpoints on this ComputeInstance.
+    :param custom_applications: Experimental. List of custom applications for the compute instance.
```
Comment (harneetvirk) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py:149]: No need of putting experimental in docstring. It should be coming from @decorator

Comment (harneetvirk) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py]: I believe, @annatisch meant to add the details about the experimental parameter in the docstring of the class.

`
class ComputeInstance(Compute):
    """Compute Instance resource.

    custom_applications is an experimental parameter.
    """
`

Comment (annatisch) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py]: This is what I had meant - but you're saying it's already handled?
I would recommend just checking the CI build artifacts to see the rendered docs and checking the parameter is documented correctly. If so, then it should be fine.

Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py]: The parameter appears to be documented correctly here: 
image

Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py]: I noticed from my previous screenshott that SetupScripts was also not public, so I have taken the liberty of adding that in as well:

image

**Review by harneetvirk — APPROVED** [review]

**Pull request merged**